components/automate-gateway/handler/applications.go

```diff
@@ -105,6 +105,22 @@ func (a *Applications) GetServicesStats(
 	return out, nil
 }
 
+func (a *Applications) GetDisconnectedServices(
+	ctx context.Context,
+	in *applications.DisconnectedServicesReq) (*applications.ServicesRes, error) {
+
+	inDomain := &applications.DisconnectedServicesReq{}
+	out := &applications.ServicesRes{}
+	f := func() (proto.Message, error) {
+		return a.client.GetDisconnectedServices(ctx, inDomain)
+	}
+	err := protobuf.CallDomainService(in, inDomain, f, out)
```

Comment:
I hate this function. But fine. 🤷‍♂

Reply:
Less hate in the world, out of curiosity, what would you recommend?

Reply:
😄 I'd prefer manually converting the incoming and outgoing protobuf messages. OR looking into the use of github.com/jhump/protoreflect to do it automatically. The "conversion" done by PB -> JSON -> PB bugs me -- even more so since grpc-gateway does JSON -> PB, too; so we end up with
```
JSON -> PB -> JSON -> PB
```

Reply:
💯agreed! We should change it.

Reply:
Also, another PR to fix this madness https://github.com/chef/automate/pull/890